tests/test_browser.py

```diff
@@ -1076,6 +1076,53 @@ def test_sdl_joystick_2(self):
     Popen([PYTHON, EMCC, os.path.join(self.get_dir(), 'sdl_joystick.c'), '-O2', '--minify', '0', '-o', 'page.html', '--pre-js', 'pre.js', '-lSDL', '-lGL']).communicate()
     self.run_browser('page.html', '', '/report_result?2')
 
+  def test_glfw_joystick(self):
```

Comment:
Added a simple interactive test (run with `python tests/runner.py interactive.test_glfw_joystick`), no graphics it only logs text of the gamepad state changes but useful for testing (already found & fixed the axes memory allocation error thanks to it)